aspnetcore/blazor/components/index.md

```diff
@@ -783,6 +783,102 @@ In the preceding example:
 > [!IMPORTANT]
 > If a Razor component defines an event that's triggered from a background thread, the component might be required to capture and restore the execution context (<xref:System.Threading.ExecutionContext>) at the time the handler is registered. For more information, see [Calling `InvokeAsync(StateHasChanged)` causes page to fallback to default culture (dotnet/aspnetcore #28521)](https://github.com/dotnet/aspnetcore/issues/28521).
 
+:::moniker-end
+
+:::moniker range=">= aspnetcore-8.0"
+
+### Dispatch exceptions to Blazor's synchronization context
+
+Use `ComponentBase.DispatchExceptionAsync` in Razor components to process exceptions thrown outside of Blazor's synchronization context. This permits the app's code to treat the exceptions as through they're lifecycle method exceptions. Thereafter, Blazor's error handling mechanisms, such as [error boundaries](xref:blazor/fundamentals/handle-errors#error-boundaries), can process the exceptions.
+
+> [!NOTE]
+> `ComponentBase.DispatchExceptionAsync` is used in Razor component files (`.razor`) that inherit from <xref:Microsoft.AspNetCore.Components.ComponentBase>. When creating components that [implement <xref:Microsoft.AspNetCore.Components.IComponent> directly](#component-classes), use `RenderHandle.DispatchExceptionAsync`.
+
+To dispatch exceptions to Blazor's synchronization context, use a `try-catch` block to pass the exception to `DispatchExceptionAsync` and await the result:
```

Comment:
This seems a bit odd because:

1. There's not necessarily any need to use `InvokeAsync` here - that depends on what code they are going to call. Whether or not `InvokeAsync` is required is unrelated to the scenario.
2. We probably don't need to explain to people how to catch exceptions - that's a basic C# topic which we can assume readers already know.

Of course you *can* use try/catch in your example, and it makes sense to do so, but it's just odd to point it out in the description as if there's some other way to get exceptions.